data-host/library/src/popup/components/SendTokens.tsx

```diff
@@ -32,6 +34,7 @@ export function SendTokens(props: { onFinish: () => void }) {
       amount={amount}
       onChangeAmount={setAmount}
       onContinue={() => setPage("confirm")}
+      onCancel={() => navigate(props.onFinish)}
```

Comment:
I don't like that this component knows about navigation. `onFinish` should be a callback and the calling component can navigate.

Reply:
It is not possible to initialise the navigate function in the Router where the SendTokens component is used. Maybe an alternative is to use Link component from the react-router library instead?

Reply:
We discussed this and can use whatever return-to-previous navigation exists. I still prefer this component not knowing about navigation and just having a callback when it is finished.